supervisely/project/project.py

```diff
@@ -4401,14 +4408,42 @@ async def _download_project_async(
     only_image_tags: Optional[bool] = False,
     save_image_info: Optional[bool] = False,
     save_images: Optional[bool] = True,
-    progress_cb: Optional[Callable] = None,
+    progress_cb: Optional[Union[tqdm, Callable]] = None,
     save_image_meta: Optional[bool] = False,
     images_ids: Optional[List[int]] = None,
     resume_download: Optional[bool] = False,
+    **kwargs,
 ):
+    """
+    Download image project to the local directory asynchronously.
+    Uses qeueu and semaphore to control the number of parallel downloads.
+    Every image goes through size check to decide if it should be downloaded in bulk or one by one.
+    Checked images are split into two lists: small and large. Small images are downloaded in bulk, large images are downloaded one by one.
+    As soon as the task is created, it is put into the queue. Workers take tasks from the queue and execute them.
+
+    """
+    # to switch between single and bulk download
+    switch_size = kwargs.get("switch_size", 1.28 * 1024 * 1024)
```

Comment:
why 1.28 ? (where did we get this number?)

Reply:
These are approximate recommended values for dividing files into large and small categories to fit within the recommended batch size. This scheme is a simple implementation of a more complex image dividing method. 
At the moment, this number is as it is, but we need to find an optimal value that will be effective in various scenarios.